### PR TITLE
Fix object numeration issue in mortar thermal action 

### LINKED ISSUE #23869: Multiple thermal pairs with mortar gap heat transfer action
## Bug Description
A user wants to leverage the `MortarGapHeatTransfer` action for multiple thermal pairs. An error message prevents the simulation from running:

```
*** ERROR ***
The following error occurred in the object "cap_hat_ModularGapConductanceConstraint", of type "ModularGapConductanceConstraint".The requested UserObject with the name "gap_flux_model_conduction_object_cap_hat_0" was not found.
```
Obviously we shouldn't be creating user objects if we can't pass their proper names to the constraints.

## Steps to Reproduce
Run the mortar gap heat transfer action with multiple thermal pairs employing the same physics (e.g. conduction or radiation).

## Impact
It prevents users from leveraging the automatic setups created by the action.

### PR DESCRIPTION
**Pull request by recuero**
Closes #23869.

### CODE REVIEW

**Review by lindsayad — APPROVED** [review]

File: modules/heat_conduction/src/actions/MortarGapHeatTransferAction.C [hunk 1]
```diff
@@ -256,22 +259,20 @@ MortarGapHeatTransferAction::addConstraints()
...
-    unsigned int radiation_index = 0;
 
     for (const auto & uo_name : _gap_flux_models)
     {
       if (uo_name == MortarGapHeatTransfer::UserObjectToBuild::CONDUCTION)
-        uoname_strings.push_back("gap_flux_model_conduction_object_" +
-                                 MooseUtils::shortName(name()) + "_" +
-                                 Moose::stringify(conduction_index++));
+        uoname_strings.push_back(
+            "gap_flux_model_conduction_object_" + MooseUtils::shortName(name()) + "_" +
```
Comment (lindsayad) [inline modules/heat_conduction/src/actions/MortarGapHeatTransferAction.C:268]: Why is the short name not enough for disambiguation? Shouldn't there be one mortar contact syntax block per primary-secondary boundary pair?

Comment (lindsayad) [reply modules/heat_conduction/src/actions/MortarGapHeatTransferAction.C:268]: If you think someone someday might want multiple radiation or conduction uos, then it makes sense to me to keep things as you've done them here. I just wanted to understand a little bit more for my own edification

Comment (lindsayad) [reply modules/heat_conduction/src/actions/MortarGapHeatTransferAction.C:268]: I was thinking; does it make more sense to have this be a class member as opposed to a file static? Again the `shortName()` grants uniqueness between mortar interfaces. The only time you have ambiguity is if you need to create multiple conduction/radiation uos on a per-mortar-interface basis, right?

Comment (recuero) [reply modules/heat_conduction/src/actions/MortarGapHeatTransferAction.C:268]: > The only time you have ambiguity is if you need to create multiple conduction/radiation uos on a per-mortar-interface basis, right?

Yes, that's what I was trying to say above. That the design should allow for having multiple uos per unit of mortar interface.
I'll create another PR using the short name.

Comment (recuero) [reply modules/heat_conduction/src/actions/MortarGapHeatTransferAction.C:268]: Counter are often static members in some module actions. I don't remember having used members to account for multiple action instantiations.

Comment (recuero) [reply modules/heat_conduction/src/actions/MortarGapHeatTransferAction.C:268]: https://github.com/idaholab/moose/pull/23876

**Pull request merged**